### ISSUE TITLE
Trying to understand the code and getting a 8x32 matrix to work with it :-)

### ISSUE THREAD

**Issue by fribse** [rnauber/ESPHomeMatrixLED#7]
Hi Richard (@rnauber) and Alessandro (@apolselli )
I have home assistant running with ESPHome and MQTT addons, so it looks like this would be a perfect match for this.
As I'm not really a programmer I'm struggling a bit, trying to understand what the code does.
I hope you can help me.
I can see that you've built quite a few effects into it.
My main goal is the text display, to have a text from a home assistant sensor showing up on the matrix (or maybe a MQTT message).
The led matrix I have is the same as some of the other posters have shown, so I guess I have to add the extra code line that you've shown because of the orientation.

I have a few questions though.
What does the MQTT do? Switch between the effects? How do I use that? ;-)
Do you have any further examples I could look at?

**Comment by rnauber** [comment]
Hi @fribse,
nice to see you are getting started with esphome.

> What does the MQTT do? Switch between the effects? How do I use that? ;-)
MQTT is the protocol that home assistant and your esp home device uses to communicate. What it is able to do is defined by your esp home yaml file.

> Do you have any further examples I could look at?

The example that fits your use case (display text from home assistant) is:
https://github.com/rnauber/ESPHomeMatrixLED/blob/master/m5atom_marquee.yaml

**Comment by fribse** [comment]
Hi @Mauber
The first matrix I had was faulty, and when I tried repairing it, I broke it completely :-)
I'm getting a couple of new ones today, looking forward to testing it.
So if I understand your example I just send a mqtt topic with device_name/text and Payload is "SomeText"
And if I use the other examples I can switch effect with device_name/effect and Payload to "Selected effect"?
Did I understand it correctly?

**Comment by mauber** [comment]
I’m sorry I’m not the “right” @mauber

 

​









Maurizio Berti
Italian Village  Developer 

	


A  Italy 



 

 


E  caseauneuro <mailto:caseauneuro@gmail.com > @gmail.com  



W   <https://casea1euro.it/> https://casea1euro.it/  


Skype maurizio berti 




 <http://www.facebook.com/casea1euro/> 

 <http://www.linkedin.com/in/maurizio-berti-80804047/> 

	

 


 <http://linkedin.com/in/maurizio-berti-80804047/>   

 

 

Da: Kenneth Fribert <notifications@github.com> 
Inviato: lunedì 5 ottobre 2020 09:48
A: rnauber/ESPHomeMatrixLED <ESPHomeMatrixLED@noreply.github.com>
Cc: mauber <maurizioberti1959@gmail.com>; Mention <mention@noreply.github.com>
Oggetto: Re: [rnauber/ESPHomeMatrixLED] Trying to understand the code and getting a 8x32 matrix to work with it :-) (#7)

 

Hi @mauber <https://github.com/mauber> 
The first matrix I had was faulty, and when I tried repairing it, I broke it completely :-)
I'm getting a couple of new ones today, looking forward to testing it.
So if I understand your example I just send a mqtt topic with device_name/text and Payload is "SomeText"
And if I use the other examples I can switch effect with device_name/effect and Payload to "Selected effect"?
Did I understand it correctly?

**Comment by fribse** [comment]
Hi @rnauber 
The first matrix I had was faulty, and when I tried repairing it, I broke it completely :-)
I'm getting a couple of new ones today, looking forward to testing it.
So if I understand your example I just send a mqtt topic with device_name/text and Payload is "SomeText"
And if I use the other examples I can switch effect with device_name/effect and Payload to "Selected effect"?
Did I understand it correctly?
